health.go

```diff
@@ -242,23 +286,48 @@ func (h *Health) startRunner(cfg *Config, ticker *time.Ticker, stop <-chan struc
 
 		h.Logger.WithFields(log.Fields{"name": cfg.Name}).Debug("Checker exiting")
 	}()
-
-	return nil
 }
 
+// resets the states in a concurrency-safe manner
 func (h *Health) safeResetStates() {
 	h.statesLock.Lock()
 	defer h.statesLock.Unlock()
 	h.states = make(map[string]State, 0)
 }
 
+// updates the check state in a concurrency-safe manner
 func (h *Health) safeUpdateState(stateEntry *State) {
 	// update states here
 	h.statesLock.Lock()
 	defer h.statesLock.Unlock()
+
+	// track failures if there is a status listener
+	if h.StatusListener != nil {
```

Comment:
Let's stick this into a helper method (ie. `h.handleStatusListener(entry *State)` -- this should keep things a bit cleaner. Also, I'd break out early to avoid the deeper if block -- ie. `if h.StatusListener == nil { return } // followed by actual logic`.